database/sql/src/main/scala/cromwell/database/slick/tables/WorkflowMetadataSummaryEntryComponent.scala

```diff
@@ -96,6 +98,9 @@ trait WorkflowMetadataSummaryEntryComponent {
       val labelsOrFilter = existsWorkflowLabels(workflowMetadataSummaryEntry, labelOrKeyValues, _ || _)
       val excludeLabelsAndFilter = existsWorkflowLabels(workflowMetadataSummaryEntry, excludeLabelAndValues, _ && _).map(v => !v)
       val excludeLabelsOrFilter = existsWorkflowLabels(workflowMetadataSummaryEntry, excludeLabelOrValues, _ || _).map(v => !v)
+      val notASubworkflowFilter: Option[Rep[Boolean]] =
+        if (includeSubworkflows) None else Some(metadataEntryExistsForWorkflowExecutionUuid(workflowMetadataSummaryEntry.workflowExecutionUuid, parentIdWorkflowMetadataKey))
+
```

Comment:
@cjllanwarne Do you think adding the filtering of subworkflows when `includeSubworkflows = false` here, might resolve the issue: #3873? Or is it unrelated?

Reply:
Great question! @rexwangcc do you still have the database of >1000 jobs to test this version against to see if anything changed, got better, or got worse?

Reply:
I feel it might make a difference, because before all the workflows were fetched which would match other filters, and then each of them would be checked for subworkflow condition if `includeSubworkflows = false` was set. Now, the subworkflows condition is also checked along with other filters, and if it is a subworkflow it won't be included in the database result itself (when `includeSubworkflows = false`). It might resolve the issue!